### PR TITLE
Remove the data generating kernel from `estimationProblem`

### PR DESCRIPTION
**Pull request by RemyDegenne**

### CODE REVIEW

File: TestingLowerBounds/Testing/Risk.lean [hunk 1]
```diff
@@ -33,92 +33,88 @@ namespace ProbabilityTheory
...
+@[simp]
+lemma risk_comp_assoc (E : estimationProblem Θ 𝒴 𝒵) (P : kernel Θ 𝒳) [IsSFiniteKernel P]
+    (κ : kernel 𝒳'' 𝒵) (θ : Θ)
+    (η : kernel 𝒳 𝒳') [IsMarkovKernel η] (η' : kernel 𝒳' 𝒳'') [IsMarkovKernel η'] :
+    risk E (η' ∘ₖ (η ∘ₖ P)) κ θ = risk E (η' ∘ₖ η ∘ₖ P) κ θ := by
+  rw [risk, risk]
+  congr 1
+  simp_rw [kernel.comp_apply]
+  congr 1
+  rw [Measure.comp_assoc]
```
Comment (LorenzoLuccioli) [inline TestingLowerBounds/Testing/Risk.lean]:
```suggestion
  rw [kernel.comp_assoc]
```

File: TestingLowerBounds/Testing/Risk.lean [hunk 2]
```diff
@@ -33,92 +33,88 @@ namespace ProbabilityTheory
...
+  _ ≤ ∫⁻ _, (⨆ θ', risk E P κ θ') ∂π := lintegral_mono (fun θ ↦ le_iSup _ _)
+  _ = ⨆ θ, risk E P κ θ := by simp
 
-lemma bayesianRisk_comap_measurableEquiv (E : estimationProblem Θ 𝒳 𝒴 𝒵) [IsSFiniteKernel E.P]
+@[simp]
+lemma bayesianRisk_comp_assoc (E : estimationProblem Θ 𝒴 𝒵) (P : kernel Θ 𝒳) [IsSFiniteKernel P]
+    (κ : kernel 𝒳'' 𝒵) (π : Measure Θ)
+    (η : kernel 𝒳 𝒳') [IsMarkovKernel η] (η' : kernel 𝒳' 𝒳'') [IsMarkovKernel η'] :
+    bayesianRisk E (η' ∘ₖ (η ∘ₖ P)) κ π = bayesianRisk E (η' ∘ₖ η ∘ₖ P) κ π := by
+  simp [bayesianRisk]
```
Comment (LorenzoLuccioli) [inline TestingLowerBounds/Testing/Risk.lean]:
```suggestion
  rw [kernel.comp_assoc]
```

File: TestingLowerBounds/Testing/Risk.lean (hunk 1 above)
Comment (LorenzoLuccioli) [reply TestingLowerBounds/Testing/Risk.lean]: Do we actually need this lemma?
It is just an application of `kernel.comp_assoc`.

File: TestingLowerBounds/Testing/Risk.lean (hunk 2 above)
Comment (LorenzoLuccioli) [reply TestingLowerBounds/Testing/Risk.lean]: Same as `risk_comp_assoc`

File: TestingLowerBounds/Testing/Risk.lean (hunk 1 above)
Comment (RemyDegenne) [reply TestingLowerBounds/Testing/Risk.lean]: Somehow I forgot about kernel.comp_assoc ? I don't know why.

**Review by LorenzoLuccioli — APPROVED** [review]
Looks good to me!

**Pull request merged**